molecule/cookiecutter/scenario/driver/ec2/{{cookiecutter.molecule_directory}}/{{cookiecutter.scenario_name}}/create.yml

```diff
@@ -29,41 +30,64 @@
     keypair_name: molecule_key
     keypair_path: "{{ lookup('env', 'MOLECULE_EPHEMERAL_DIRECTORY') }}/ssh_key"
   tasks:
+    - name: Check subnet
+      ec2_vpc_subnet_facts:
+        region: "{{ item.region|default(default_region) }}"
+        filters:
+          subnet-id: "{{ item.vpc_subnet_id }}"
+      with_items: "{{ molecule_yml.platforms }}"
+      register: subnet_facts
+
+    - name: Populate subnet vpc
+      set_fact:
+        subnet_dict: "{{ subnet_dict|default({}) | combine({ item.subnets[0].id: item.subnets[0].vpc_id }) }}"
+      with_items: "{{ subnet_facts.results }}"
+
     - name: Create security group
       ec2_group:
-        name: "{{ security_group_name }}"
-        description: "{{ security_group_name }}"
+        name: "{{ security_group_name_prefix }}-{{ subnet_dict[item.vpc_subnet_id] }}"
+        description: "{{ security_group_description }}"
         rules: "{{ security_group_rules }}"
         rules_egress: "{{ security_group_rules_egress }}"
+        vpc_id: "{{ subnet_dict[item.vpc_subnet_id] }}"
+        region: "{{ item.region|default(default_region) }}"
+      with_items: "{{ molecule_yml.platforms }}"
 
     - name: Test for presence of local keypair
       stat:
-        path: "{{ keypair_path }}"
```

Comment:
As far as I remember keypairs are global and not region specified, it makes no sense to add region to a key.

Reply:
unfortunately, key pairs are region-specific 😞